meilisearch-http/tests/auth/api_keys.rs

```diff
@@ -936,19 +1160,20 @@ async fn patch_api_key_indexes() {
 
     let (response, code) = server.add_api_key(content).await;
     // must pass if add_valid_api_key test passes.
-    assert_eq!(code, 201);
+    assert_eq!(201, code, "{:?}", &response);
     assert!(response["key"].is_string());
     assert!(response["createdAt"].is_string());
     assert!(response["updatedAt"].is_string());
 
-    let key = response["key"].as_str().unwrap();
+    let uid = response["uid"].as_str().unwrap();
     let created_at = response["createdAt"].as_str().unwrap();
     let updated_at = response["updatedAt"].as_str().unwrap();
 
     let content = json!({ "indexes": ["products", "prices"] });
 
     thread::sleep(time::Duration::new(1, 0));
-    let (response, code) = server.patch_api_key(&key, content).await;
+    let (response, code) = server.patch_api_key(&uid, content).await;
+    assert_eq!(200, code, "{:?}", &response);
```

Comment:
We should probably return an error when a user tries to change values that must not change, we should return [a 403 HTTP response](https://web.archive.org/web/20200110002346/https://httpstatuses.com/403) instead of making him think the change succeeded. @gmourier 

The same applies when we try to change the actions of the API keys.

Reply:
It could also be a `422: Unprocessable Entity` right? That's why I kept a `400 Bad Request` code that can work if a field can't be updated because there are several possibilities of HTTP code and I didn't find a proper answer. 🧐

What do you think about keeping `400 Bad Request`, doesn't the message bring what is needed in terms of understanding?

Reply:
Right now we do not return any error when a user tries to change the actions and/or the indexes restrictions of a key. It doesn't really matter to me the HTTP code we return to the user in case of an error, a good 400 Bad Access is really good too!

Reply:
Sure, I missed adding the errors! Thanks @Kerollmops!